Sources/SourceKitLSP/Swift/SwiftLanguageService.swift

```diff
@@ -165,6 +162,15 @@ public actor SwiftLanguageService: LanguageService, Sendable {
 
   private var diagnosticReportManager: DiagnosticReportManager!
 
+  var documentManager: DocumentManager {
+    get throws {
+      guard let sourceKitLSPServer = self.sourceKitLSPServer else {
+        throw NSError(domain: "SourceKitLSPErrorDomain", code: 1001, userInfo: [NSLocalizedDescriptionKey: "Cannot get document manager because sourceKitLSPServer was deallocated"])
```

Comment:
We generally don’t use `NSError` in SourceKit-LSP. I would either:
- Define a custom error type that we can throw here or
- follow the precedence set [here](https://github.com/apple/sourcekit-lsp/blob/ed4eea334cff2968dae14a9170817fd7c98a6019/Sources/SourceKitLSP/Swift/SwiftLanguageService.swift#L933) and return `ResponseError.unknown("Connection to the editor closed")`

Reply:
Yeah I feel like ResponseError would be suffice here, having a custom error seems a bit of an overkill for now